plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -127,15 +129,40 @@ public PluginManager(Config cfg) {
         skipFailedPlugins = cfg.isSkipFailedPlugins();
         hashFunction = cfg.getHashFunction();
         httpClient = null;
+        try{
+            userAgentInformation = this.getUserAgentInformation();
+        }
+        catch(UserAgentInformationException e){
+            logVerbose("Unable to get the UserAgentInformation");
+        }
     }
 
+    private String getUserAgentInformation() throws UserAgentInformationException{
+        Properties properties = new Properties();
+        ClassLoader propertiesClassLoader = this.getClass().getClassLoader();
+        try(InputStream propertiesStream = propertiesClassLoader.getResourceAsStream("Version.properties")){
+            if(propertiesStream != null){
+                properties.load(propertiesStream);
+                userAgentInformation = properties.getProperty("project.artifactId") +"/"+properties.getProperty("project.version");
```

Comment:
there should be a variable internal to this method, this assignment is modifying the state outside of this method